plugin/api.go

```diff
@@ -480,6 +480,36 @@ type API interface {
 	//
 	// Minimum server version: 5.7
 	SendMail(to, subject, htmlBody string) *model.AppError
+
+	// CreateBot creates the given bot and corresponding user.
+	//
+	// Minimum server version: 5.8
+	CreateBot(bot *model.Bot) (*model.Bot, *model.AppError)
+
+	// PatchBot applies the given patch to the bot and corresponding user.
+	//
+	// Minimum server version: 5.8
+	PatchBot(userId string, botPatch *model.BotPatch) (*model.Bot, *model.AppError)
```

Comment:
Are we sure we want to say userId here? It's a bit confusing. I understand where it's coming from, but maybe just `id` would be clearer? `1/5`

Reply:
I had been inclined to adopt the term `botUserId` for maximum clarity, given that bots don't have an `Id` -- they only have `UserId`. I think I'll do that now given the feedback.